src/pymor/reductors/h2.py

```diff
@@ -831,3 +831,155 @@ def _compute_conv_crit(self, rom, conv_crit, it):
 
         self.conv_crit.append(dist)
         self.logger.info(f'Convergence criterion in iteration {it + 1}: {dist:e}')
+
+
+class VectorFittingReductor(BasicObject):
+    """Vector fitting reductor.
+
+    Only for single-input single-output (SISO) systems.
+
+    Parameters
+    ----------
+    s
+        Sampling points in the complex plane as a 1D |NumPy array|.
+    Hs
+        Transfer function values at the sampling points `s` as a 1D |NumPy array|.
+        Alternatively, |TransferFunction| or `Model` with `transfer_function` attribute.
+    weights
+        Weights in the weighted least squares error as a 1D |NumPy array|.
+        If not given, it is set to a vector of ones.
+    conjugate
+        Whether to include conjugated data.
+    """
+
+    def __init__(self, s, Hs, weights=None, conjugate=True):
+        assert isinstance(s, np.ndarray)
+        assert s.ndim == 1
+
+        if hasattr(Hs, 'transfer_function'):
+            Hs = Hs.transfer_function
+        if isinstance(Hs, TransferFunction):
+            Hs = np.array([Hs.eval_tf(si).squeeze() for si in s])
```

Comment:
I believe we have the informal convention in pyMOR that instantiating object should be cheap. Thus, it might make sense to move the evaluation of the transfer function in a separate method, which is then called in the first call to `reduce`.

Reply:
Might get a bit messy when `conjugate is True`, however.

Reply:
Yes, that's a good point. I copied from [PAAA](https://github.com/pymor/pymor/blob/c912818b0fa5b232cf7143119aa041998c9589d8/src/pymor/reductors/aaa.py#L69) and [Loewner](https://github.com/pymor/pymor/blob/c912818b0fa5b232cf7143119aa041998c9589d8/src/pymor/reductors/loewner.py#L49), so I guess they should also then be modified...

Reply:
Probably. Do you want to do it today? Otherwise, we could also merge as is and open an issue to not forget about it.

Reply:
I tried, and it's doable for PAAA, but Loewner is cacheable, and it seems ugly to use a bunch of private attributes...

Reply:
Ok, then let's merge as is for now.